### PR TITLE
Add draggable content component

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/DraggableContent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+import { GrDrag } from 'react-icons/gr';
```
Comment (tnagorra) [inline src/components/DraggableContent/index.tsx:3]: Are we going to use different icon styles throughout the deep-ui?

File: src/components/RawInput/index.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { _cs } from '@togglecorp/fujs';
 
-import type { UiMode } from '../UiModeContext';
-import useUiModeClassName from '../../hooks/useUiModeClassName';
+import type { UiMode } from '#components/UiModeContext';
+import useUiModeClassName from '#hooks/useUiModeClassName';
```
Comment (tnagorra) [inline src/components/RawInput/index.tsx]: Please use relative import

File: src/stories/DraggableContent.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { withDesign } from 'storybook-addon-designs';
+import { Story } from '@storybook/react/types-6-0';
+
+import DraggableContent, { Props as DraggableContentProps } from '#components/DraggableContent';
```
Comment (tnagorra) [inline src/stories/DraggableContent.stories.tsx:5]: Please use relative import

File: src/components/DraggableContent/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+import ElementFragments from '#components/ElementFragments';
```
Comment (tnagorra) [inline src/components/DraggableContent/index.tsx]: Please use relative import

File: src/components/DraggableContent/index.tsx (hunk 1 above)
Comment (frozenhelium) [reply src/components/DraggableContent/index.tsx:3]: By the order of preference
1. Io5
2. Io4
3. Other

**Review by samshara — APPROVED** [review]

File: src/stories/DraggableContent.stories.tsx (hunk 3 above)
Comment (tnagorra) [reply src/stories/DraggableContent.stories.tsx:5]: Sorry sir, not here!

File: src/components/DraggableContent/index.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+type SerializableValue = Record<string, unknown>;
+
+export interface Props {
+    className?: string;
+    children: React.ReactNode;
+    elementRef?: DivElementProps['ref'];
+    value?: SerializableValue;
+    dropEffect?: 'copy' | 'move' | 'link' | 'none';
+    onDragStart?: (value?: SerializableValue) => void;
```
Comment (tnagorra) [inline src/components/DraggableContent/index.tsx]:
```suggestion
    onDragStart?: (value: SerializableValue | undefined) => void;
```

File: src/components/DropContainer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import { _cs } from '@togglecorp/fujs';
+
+import type { Props as DraggableContentProps } from '../DraggableContent';
+
+import styles from './styles.css';
+
+export interface Props {
+    className?: string;
+    children?: React.ReactNode;
+    onDrop?: (value?: DraggableContentProps['value']) => void;
```
Comment (tnagorra) [inline src/components/DropContainer/index.tsx]:
```suggestion
    onDrop?: (value: DraggableContentProps['value'] | undefined) => void;
```

File: src/stories/DragAndDrop/DragAndDrop.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        </DraggableContent>
+        <DropContainer className={styles.dropContainer}>
+            Drop container
+        </DropContainer>
+    </div>
+);
+
+export const Default = Template.bind({});
+Default.args = {
+    children: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla sed convallis quam, quis molestie nisi. Integer fringilla maximus tellus at aliquam. Nunc ac turpis non elit placerat luctus. Mauris vehicula, dui vitae feugiat malesuada, diam elit porttitor tellus, ut ultricies nibh est at ante. Maecenas congue congue nulla quis feugiat. Etiam porta volutpat mollis.',
```
Comment (tnagorra) [inline src/stories/DragAndDrop/DragAndDrop.stories.tsx]: This is not necessary!

File: src/stories/DraggableContent.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (tnagorra) [inline src/stories/DraggableContent.stories.tsx:1]: Isn't this repeated?

**Review by tnagorra — CHANGES_REQUESTED** [review]
It would be better if we could interact with the contens of the draggable container.

Also, we need to style the draggable/droppable container.

Also, there's some problem with the drop-container. Once we drag over it, the state doesn't change on drop.
Also, can we make a global state so that once a drag is started, we can change styling for all droppable content.

**Pull request merged**